tests/TestCase/ORM/TableTest.php

```diff
@@ -4284,6 +4284,94 @@ public function testSaveCorrectPrimaryKeyType()
     }
 
     /**
+     * Tests the loadInto() method
+     *
+     * @return void
+     */
+    public function testLoadIntoEntity()
+    {
+        $table = TableRegistry::get('Authors');
+        $table->hasMany('SiteArticles');
+        $articles = $table->hasMany('Articles');
+        $articles->belongsToMany('Tags');
+
+        $entity = $table->get(1);
+        $result = $table->loadInto($entity, ['SiteArticles', 'Articles.Tags']);
+        $this->assertSame($entity, $result);
+
+        $expected = $table->get(1, ['contain' => ['SiteArticles', 'Articles.Tags']]);
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * Tests that it is possible to pass conditions and fields to loadInto()
+     *
+     * @return void
+     */
+    public function testLoadIntoWithConditions()
+    {
+        $table = TableRegistry::get('Authors');
+        $table->hasMany('SiteArticles');
+        $articles = $table->hasMany('Articles');
+        $articles->belongsToMany('Tags');
+
+        $entity = $table->get(1);
+        $options = [
+            'SiteArticles' => ['fields' => ['title', 'author_id']],
+            'Articles.Tags' => function ($q) {
+                return $q->where(['Tags.name' => 'tag2']);
+            }
+        ];
+        $result = $table->loadInto($entity, $options);
+        $this->assertSame($entity, $result);
+        $expected = $table->get(1, ['contain' => $options]);
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * Tests loadInto() with a belongsTo association
+     *
+     * @return void
+     */
+    public function testLoadBelognsTo()
+    {
+        $table = TableRegistry::get('Articles');
+        $table->belongsTo('Authors');
+
+        $entity = $table->get(2);
+        $result = $table->loadInto($entity, ['Authors']);
+        $this->assertSame($entity, $result);
+
+        $expected = $table->get(2, ['contain' => ['Authors']]);
+        $this->assertEquals($expected, $entity);
+    }
+
+    /**
+     * Tests that it is possible to post-load associations for many entities at
+     * the same time
+     *
+     * @return void
+     */
+    public function testLoadIntoMany()
+    {
+        $table = TableRegistry::get('Authors');
+        $table->hasMany('SiteArticles');
+        $articles = $table->hasMany('Articles');
+        $articles->belongsToMany('Tags');
+
+        $entities = $table->find()->toList();
```

Comment:
Can loadInto() accept a collection? There aren't tests covering that from what I can see.